### PR TITLE
Add dynamic generation of `robots.txt`

### PR DESCRIPTION
**Pull request by 3w36zj6**
#### Description (required)

The current documentation of [@astrojs/sitemap](https://docs.astro.build/en/guides/integrations-guide/sitemap/) does not provide a setup for generating a dynamic `robots.txt`. 

It introduces the following basic configuration for `robots.txt`:

```robots
User-agent: *
Allow: /

Sitemap: https://<YOUR SITE>/sitemap-index.xml
```

Ideally, we would like to reference the `site` value set in `astro.config`, but it only suggests a method for hardcoding this URL.

### CODE REVIEW

File: src/content/docs/en/guides/integrations-guide/sitemap.mdx
Comment (liruifengv) [inline src/content/docs/en/guides/integrations-guide/sitemap.mdx:122]: We do not recommend using the first person in documents.

You can find writing and style Guide in [Astro Docs Docs](https://contribute.docs.astro.build/guides/writing-style/#writing-style)

**Review by liruifengv — COMMENTED** [review]
Thanks for your contribution. @3w36zj6 
I made a suggestion. Also, is `dedent` required here?

**Comment by liruifengv** [comment]
My website example is

```ts
import type { APIRoute } from "astro";

const robots = `
User-agent: *
Disallow:

User-agent: *
Allow: /

Sitemap: ${new URL("sitemap-index.xml", import.meta.env.SITE).href}
`.trim();

export const GET: APIRoute = () =>
  new Response(robots, {
    headers: { "Content-Type": "text/plain" },
  });

```

**Review by VoxelMC — APPROVED** [review]
Great idea, LGTM!

File: src/content/docs/en/guides/integrations-guide/sitemap.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/integrations-guide/sitemap.mdx:122]:
```suggestion
To generate `robots.txt` dynamically, add a file named `robots.txt.ts` with the following code:
```
We can make this even shorter and more direct!

**Review by sarah11918 — COMMENTED** [review]
Thank you @3w36zj6 , this looks really helpful and is a wonderful contribution to the docs!

Thank you for working with others on the team to get this polished! And thank you @liruifengv for the wonderful review!

I'm going to lightly edit some text that is more to our current style. (I could probably edit more on this whole page, it was written a long time ago!) and then we'll wait for final approvals and are very happy to welcome you to Team Docs! :partying_face:

**Comment by sarah11918** [comment]
@liruifengv Looks good to me. Just waiting for you to approve the final version and I'll merge it!  🫡

**Review by liruifengv — APPROVED** [review]
LGTM!

**Review by ElianCodes — APPROVED** [review]

**Pull request merged**
